solr/core/src/java/org/apache/solr/core/PluginBag.java

```diff
@@ -138,14 +140,27 @@ static void initInstance(Object inst, PluginInfo info) {
       log.debug("{} : '{}' created with startup=lazy ", meta.getCleanTag(), info.name);
       return new LazyPluginHolder<T>(meta, info, core, core.getResourceLoader(), false);
     } else {
-      T inst = core.createInstance(info.className, (Class<T>) meta.clazz, meta.getCleanTag(), null, core.getResourceLoader());
-      initInstance(inst, info);
-      return new PluginHolder<>(info, inst);
+      if (info.pkgName != null) {
```

Comment:
minor: please tack on another "else if" from above rather than indenting further needlessly.  Also for consistency it's debatable if getPackagePluginHolder needs its own method vs. inlining.   There is no getRuntimeLibPluginHolder after all.

Reply:
inlined